File: 1395-count-number-of-teams/1395-count-number-of-teams.java
class Solution {
    public int numTeams(int[] rating) {
        if(rating.length < 3){
            return 0;
        }
        
        int n = rating.length;
        int res = 0;
        
        HashMap<Integer, Integer> asc = new HashMap<Integer, Integer>();
        HashMap<Integer, Integer> desc = new HashMap<Integer, Integer>();
        
        for(int i = 0; i < n; i++){
            for(int j = i+1; j < n; j++){
                if(rating[i] < rating[j]){
                    asc.put(i, asc.getOrDefault(i, 0) + 1);
                }else{
                    desc.put(i, desc.getOrDefault(i, 0) + 1);
                    
                }
            }
        }
        
        for(int i = 0; i < n - 2; i++){
            for(int j = i+1; j < n - 1; j++){
                if(rating[i] < rating[j]){
                    res += asc.getOrDefault(j, 0);
                }else{
                    res += desc.getOrDefault(j, 0);
                }
            }
        }
        
        return res;
        
    }
}
           

File: 1395-count-number-of-teams/1395-count-number-of-teams.py
class Solution:
    def numTeams(self, rating: List[int]) -> int:
        ans=0
        #[2,5,3,4,1]
        # 3 --> sol=1 , lol=1
        # 3 --> sor =1 , lor=1
        # asc=sol*lor
        # desc=sor*lol
        
        for i in range(len(rating)):
            sol=0;lol=0
            for j in range(i):
                if rating[j]<rating[i]:
                    sol+=1
                if rating[j]>rating[i]:
                    lol+=1
            sor=0;lor=0
            for j in range(i+1,len(rating)):
                if rating[j]<rating[i]:
                    sor+=1
                if rating[j]>rating[i]:
                    lor+=1
            a=sol*lor
            d=sor*lol
            ans+=a+d
        return ans